rudder-web/src/test/scala/com/normation/rudder/web/rest/RestTestSetUp.scala

```diff
@@ -0,0 +1,136 @@
+import org.specs2.matcher.MatchResult
+import org.specs2.mutable._
+import org.specs2.runner._
+
+import net.liftweb.common.Full
+import net.liftweb.http.LiftRules
+import net.liftweb.http.LiftRulesMocker
+import net.liftweb.http.PlainTextResponse
+import net.liftweb.http.Req
+import net.liftweb.mocks.MockHttpServletRequest
+import net.liftweb.mockweb.MockWeb
+import com.normation.utils.StringUuidGeneratorImpl
+import com.normation.cfclerk.services.UpdateTechniqueLibrary
+import com.normation.eventlog.EventActor
+import com.normation.eventlog.ModificationId
+import com.normation.cfclerk.services.TechniquesLibraryUpdateNotification
+import net.liftweb.common.Box
+import com.normation.cfclerk.services.TechniquesLibraryUpdateType
+import com.normation.cfclerk.domain.TechniqueName
+import net.liftweb.http.LiftResponse
+import net.liftweb.util.NamedPF
+import net.liftweb.http.S
+import net.liftweb.common.EmptyBox
+
+
+/*
+ * This file provides all the necessary plumbing to allow test REST API.
+ *
+ * Also responsible for setting up datas and mock services.
+ */
+object RestTestSetUp {
+
+  val uuidGen = new StringUuidGeneratorImpl()
+  val reloadTechniques = new RestTechniqueReload(new UpdateTechniqueLibrary() {
+    def update(modId: ModificationId, actor:EventActor, reason: Option[String]) : Box[Map[TechniqueName, TechniquesLibraryUpdateType]] = {
+      Full(Map())
+    }
+    def registerCallback(callback:TechniquesLibraryUpdateNotification) : Unit = {}
+  }, uuidGen)
+
+  val liftRules = {
+    val l = new LiftRules()
+    l.statelessDispatch.append(RestStatus)
+    l.statelessDispatch.append(reloadTechniques)
+    //TODO: add all other rest classes here
+    l
+  }
+
+  /*
+   * Correctly build and scope mutable things to use the request in a safe
+   * way in the context of LiftRules.
+   */
+  def doReq[T](mockReq: MockHttpServletRequest)(tests: Req => MatchResult[T]) = {
+    LiftRulesMocker.devTestLiftRulesInstance.doWith(liftRules) {
+      MockWeb.useLiftRules.doWith(true) {
+        MockWeb.testReq(mockReq)(tests)
+      }
+    }
+  }
+
+  /**
+   * Execute the request and get the response.
+   * The request must be a stateless one, else a failure
+   * will follow.
+   */
+  def execRequestResponse[T](mockReq: MockHttpServletRequest)(tests: Box[LiftResponse] => MatchResult[T])= {
+    doReq(mockReq){ req =>
+      //the test logic is taken from LiftServlet#doServices.
+      //perhaps we should call directly that methods, but it need
+      //much more set-up, and I don't know for sure *how* to set-up things.
+      NamedPF.applyBox(req, LiftRules.statelessDispatch.toList).map(_.apply() match {
+        case Full(a) => Full(LiftRules.convertResponse((a, Nil, S.responseCookies, req)))
+        case r => r
+      }) match {
+        case Full(x)      => tests(x)
+        case eb: EmptyBox => tests(eb)
+      }
+    }
+  }
+
+
+  def GET(path: String) = {
+    val mockReq = new MockHttpServletRequest("http://localhost:8080")
```

Comment:
AFAIK, the server/port is only used to build the fake Request/Response, not to actually send anything on that URL. It could be anything, I believe.